### ISSUE TITLE
Two enums for stop reason.

### ISSUE THREAD

**Issue by nuid64** [sztomi/dap-rs#22]
There is `StopReason` and `StoppedEventReason` enumerations in `dap/src/types.rs` and both are almost identical. Difference is `StoppedEventReason` contains `String(String)` variant, so their `FromStr` implementations differs, too. Neither of them is converted into other. I think `StopReason` sounds better so I suggest adding `Other(String)` variant to it and remove `StoppedEventReason`.

**Comment by sztomi** [comment]
That's a good catch, however, it is probably best to keep the other one for consistency. This type is not part of the specification, it is one of the _"union of literals and string"_ type that is in the specification here: https://microsoft.github.io/debug-adapter-protocol/specification#Events_Stopped

Since there are no unnamed union types in Rust, I had to come up with a suitable name for it. 

It seems that I forgot to remove / rename `StopReason` after I adopted the convention of prefixing the type with the enclosing event structure. Similarly, there are types such as `ModuleEventReason`, `OutputEventCategory` etc. that represent similar union types.
